### DISCUSSION TITLE
New Environment Library

### DISCUSSION THREAD

**Discussion by bap14** [wardenenv/warden#793] [Ideas]
The idea here is to take our current environments and move them to another repository, or alternative location. Then environments are either cached locally or retrieved on-demand. 

Things to consider about not distributing these with the full warden release:
- Flexibility
-- Environments could be added or updated without requiring a new Warden release
- Security
-- How do we ensure the security of the downloaded environment isn't tampered with?
- Distribution
-- Clone the repository?
-- Create releases that are downloaded and extracted locally?

My initial thoughts are:
- Move the environments to a separate repository: `wardenenv/environments`
- Add a step during install to clone the environments repository to the `${WARDEN_HOME}/environments` directory
- Add a command to warden to update the environments (`git pull`)
- Automatically update the environments when running `warden svc up`
- Optional: Periodically run an environment update
- Adjust `warden env-init` to use cloned repo files
- Adjust `warden env up` to use cloned repo files

Would love feedback on the approach, or if there are things missing or other things that may need to be included here.

**Comment by navarr** [comment]
A very simple solution would be to just make the environments their own homebrew repo. 

Though that wouldn't play super nice with people not using homebrew. 

I do like the idea, but I don't find it super important either. We can release as many tiny versions as we want because people don't generally want to be on older versions of Warden. That might not always be the case, and it is certainly true that people do stay on older versions just out of not updating. 

What I think is more worthwhile of attention for this kind of use case is to expand the ability of environments for non-Warden maintained versions. Warden has always been part and parcel focused on Magento, with other environments submitted getting a kind of meh because I don't work with those and I'll be very bad about keeping them up to date. 

Having some way to pull in environment types could make Warden more useful without increasing maintenance
